### PR TITLE
STN-351: QA Feedback 16

### PR DESCRIPTION
**Pull request by heatherdesigns**
# [STN-372](https://sparkbox.atlassian.net/browse/STN-351) READY FOR REVIEW

## Summary
Resolve issues discovered during the Sprint 16 QA Feedback session listed in STN-351

## Need Review By (Date)
5/26

## Urgency
medium

## Steps to Test
1. In the CLI run `npm run test` and confirm all of our Sass tests pass.
<br>

2. Image Styles on node pages
- [x] In the top section of the node detail page, images should maintain their size and aspect-ratio as defined by the image style. The image size should remain fixed and the text region should be flexible. Pages to review:
* http://economics.suhumsci.loc/events/qa-events-all-fields-filled
* http://economics.suhumsci.loc/news/qa-joeys-test-news-item
* http://economics.suhumsci.loc/people/qa-person-all-fields-filled
* http://economics.suhumsci.loc/events/gsb-organizational-behavior-seminars/gsb-organizational-behavior-seminars-13
<br>

3. Pills 
Find a page that is currently using pills or update a page such as http://economics.suhumsci.loc/people/faculty. 
* Update the view so that `Research Areas` Multiple Field Settings > Display Type uses an `ol` and in Style Settings > Customize field HTML add the class of `hb-pill-link-list`. 
Improve the color contrast of pills in Colorful
* Update `Fields of Interest` Multiple Field Settings > Display Type to use an `ul` and in Style Settings > Customize field HTML add the class of `hb-pill-list`.
- [x] Confirm that unordered lists do not display bullets before the pill. See screenshot in [STN-351](https://sparkbox.atlassian.net/browse/STN-351) for clarification.

- [x] Improve the color contrast of pills in Colorful
Review the colors of the pills in each color pairing (Ocean, Mountain, and Cardinal). Confirm the color contrast for each has improved and that they meet [WCAG AA](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) guidelines.
<br>

4. More link
- [x] Review more link when located in the local footer dark variant. Confirm color contrast improvement and that it meets WCAG AA guidelines.
<br>

5. Row Background Color
Create a page filled for testing content within rows that have full width and limited width wells. The the screenshot for `row background color` in JIRA STN-351 as an example. Create the following scenarios for full width and limited width wells:
- [x] Views without a title --> look for adequate spacing on top
- [x] horizontal card postcard --> look for adequate spacing on top
- [x] accordions --> Make sure they span the width of the container and the plus sign does not overlap the text. I also added a background-color to the accordion summary as it was the background-color of the well.
- [x] hero --> Confirm they span the width of the container. I also made spacing adjustments for consistency.
- [x] carousel --> I made spacing adjustments for consistency and changed the background color of the grey block behind the image because it was the same color as the well. 
<br>

6. Mixing bullets and numbers in WYSIWYG results
- [x] Go to https://economics-prod.stanford.edu/graduate/phd-degree and confirm that the bullets and numbers within the `General Timeline` lists do not overlap upon each other.
<br>

7. Date Stacked Horizontal Card
- [x] Go to http://economics.suhumsci.loc/seminars-workshops/upcoming-seminars-and-events and confirm that the date, location, and speaker icon display for highlighted events and upcoming seminars.
<br>

8. Mailto links on postcards render the envelope icon
- [x] This was fixed in QA Feedback Sprint 15 work. Confirm fix by going to https://economics-dev.stanford.edu/undergraduate/honors-program. Scroll towards the bottom of the page and look for the `How to Apply` postcard.
<br>

9. More Link Traditional Arrow not quite centered (it is a tad too high)
- [x] On the Traditional theme, confirm the alignment of the arrows on vertical linked cards and the more link has improved.
<br>

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/scss/components/_card.scss [hunk 1]
```diff
@@ -147,14 +147,14 @@
 
     &--date {
       // only apply calendar icon when the label exists
-      .views-label-field-hs-event-date + .field-content {
+      [class*="views-label-field-hs-event-date"] + .field-content {
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_card.scss:150]: We are using classes generated by Drupal for the styling of the icons. Targeting `.views-label-field-hs-event-date` is fine for the highlight events but the icons were not displaying for the list of upcoming seminars. I discovered upcoming seminars has a different class name of `.views-label-field-hs-event-date-5`. I'm unsure how Drupal generates the `-5` and assume this number could be different when the date stacked horizontal pattern is used on another view. The change introduced was the only way I could think of targeting all potential scenarios.

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.row.scss [hunk 2]
```diff
@@ -7,6 +7,49 @@
...
+          }
+        }
+
+        .field-media-image {
+          margin-bottom: hb-calculate-rems(28px);
+        }
+      }
+    }
+
+    // When there is only one accordion in the row, make it grow to the entire width. This property is not needed on accordions when they are placed in a row with other items.
```
Comment (kaseybon) [inline docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.row.scss:55]: 👏

**Comment by kaseybon** [comment]
There are some issues with the image overlapping the on pages with 3 columns:
http://economics.suhumsci.loc/people/christopher-makler

Screen Shot 2020-05-26 at 9 01 20 AM

**Comment by heatherdesigns** [comment]
## Issue discovered while browser testing:

Hero in row looks fine in Safari, but doesn't expand the full width in Chrome as some breakpoints.

safari left vs chrome right

Update: Now I see if in Safari too. 😳

**Comment by heatherdesigns** [comment]
I added the `word-break` property to allow the content beside images to collapse as needed. Unfortunately this could get ugly with large images. In my local environment I updated the image styles to used `Small Square 200 x 200` for the person node detail pages because of the 2 column column. I will share this at demo.

faculty node

**Review by silverli — CHANGES_REQUESTED** [review]
one last small thing I noted is that for an accordion with a very long title, it needs bit more padding around the arrow so they dont overlap

Screen Shot 2020-05-26 at 12 19 54 PM

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.row.scss [hunk 3]
```diff
@@ -27,6 +27,10 @@
         details {
           margin: 0;
           @include hb-global-color('background-color', 'white');
+
+          summary {
+            padding-right: hb-calculate-rems(52px);
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.row.scss:32]: 👍

**Review by silverli — APPROVED** [review]
Looks great!! Awesome and thorough job on all this!

**Pull request merged**